### PR TITLE
Added inventory sync functionality to broker

### PR DESCRIPTION
**Pull request by JacobCallahan**
+ Added the ability to query actions from providers using nick-help
+ Misc enhancements and tweaks including improving inventory host removal

### CODE REVIEW

**Comment by JacobCallahan** [comment]
@ColeHiggins2 @kedark3 please review

**Comment by JacobCallahan** [comment]
@Ichimonji10 feel free to check this over as well (now that you're an official broker contributor!)

**Comment by omaciel** [comment]
Bait and switch :)

**Comment by JacobCallahan** [comment]
force push was to update inventory put to account for hosts that don't have a hostname (powered off in rhv)

File: broker/providers/ansible_tower.py [hunk 1]
```diff
@@ -134,12 +152,28 @@ def exec_workflow(self, **kwargs):
             return self._merge_artifacts(job)
         return job
 
+    def get_inventory(self, user=None):
+        """Compile a list of hosts based on any inventory a user's name is mentioned"""
```
Comment (kedark3) [inline broker/providers/ansible_tower.py:156]: So this is going to look at all of the inventories in tower for username?

Comment (JacobCallahan) [reply broker/providers/ansible_tower.py:156]: a specific user can be used, yes

File: broker/broker.py [hunk 2]
```diff
@@ -100,6 +105,36 @@ def checkin(self, host=None):
             self._hosts.remove(host)
             helpers.update_inventory(remove=host.hostname)
 
+    @staticmethod
+    def sync_inventory(provider):
+        """Aquire a list of hosts from a provider and update our inventory"""
```
Comment (omkarkhatavkar) [inline broker/broker.py:110]: 'Aquire' typo

Comment (JacobCallahan) [reply broker/broker.py:110]: ahhhh nice!!!

File: broker/providers/ansible_tower.py [hunk 3]
```diff
@@ -134,12 +152,28 @@ def exec_workflow(self, **kwargs):
...
+        user = user if user else UNAME
```
Comment (mshriver) [inline broker/providers/ansible_tower.py:157]: nit-picking, but `user = user or UNAME`

File: README.md [hunk 4]
```diff
@@ -60,3 +60,15 @@ If supported by your chosen provider, nick-help will display the additional argu
...
+Additionally, if you're unfamiliar with what actions are supported by your provider, you can get a list by passing the provider's name.
+```
+broker nick-help --provider AnsibleTower
+```
+
+**Run arbitrary actions**
+If a provider action doesn't result in a host creation/removal, Broker allows you to execute that action as well. There are a few output options available as well.
+```
+broker execute --help
+broker execute --workflow my-awesome-workflow --aditional-arg True
```
Comment (bherrin3) [inline README.md:72]: `
```suggestion
broker execute --workflow my-awesome-workflow --additional-arg True
```

File: broker/providers/ansible_tower.py (hunk 3 above)
Comment (JacobCallahan) [reply broker/providers/ansible_tower.py:157]: i'll include that in the next pr

**Pull request merged**